### ISSUE TITLE
bun:test expect is sync and doesn't work with .rejects

### ISSUE THREAD

**Issue by moderj** [oven-sh/bun#4909]
### What version of Bun is running?

1.0.0+822a00c4d508b54f650933a73ca5f4a3af9a7983

### What platform is your computer?

Linux 5.15.90.1-microsoft-standard-WSL2 x86_64 x86_64

### What steps can reproduce the bug?

create an async function that should throw an error. 
using expect with .rejects should start with await before the expect as said in the jest api
[jest-api](https://jestjs.io/docs/asynchronous#promises)

![image](https://github.com/oven-sh/bun/assets/118797659/e5f9032c-9e5d-4291-9347-5934d24fd29c)

### What is the expected behavior?

the test should run successfully with this syntax as bun should be compatible with jest api and .rejects also appear to be implemented in bun docs.

### What do you see instead?

![image](https://github.com/oven-sh/bun/assets/118797659/d6218bd8-a2c9-42af-baf0-331ba4357109)

await has no effect because the bun's type of `expect` is: 

![image](https://github.com/oven-sh/bun/assets/118797659/d39c9c0c-8b04-420e-b684-a01a3f9df47d)

The error I get: 

![image](https://github.com/oven-sh/bun/assets/118797659/2735b772-0a44-430e-bbcf-b7bdacd527e4)

**Comment by strager** [comment]
I copied [Jest's example of `expect().rejects.toThrow`](https://jestjs.io/docs/expect#rejects) (linked by [Bun's documentation](https://bun.sh/docs/test/writing#matchers)). I get a different error than you, but the root cause might be the same:
```javascript
test('rejects to octopus', () => {
  // make sure to add a return statement
  return expect(Promise.reject(new Error('octopus'))).rejects.toThrow(
    'octopus',
  );
});
```
```shell
$ bun test lol.test.mjs
bun test v1.0.1 (31aec4eb)

lol.test.mjs:
error: Expected value must be a function
✗ rejects to octopus [0.09ms]

 0 pass
 1 fail
 1 expect() calls
Ran 1 tests across 1 files. [124.00ms]
```

**Comment by moderj** [comment]
I also believe that the root cause is the same, tried multiple implementations and also ran into your error while trying. The cause is that expect can't handle `async` calls inside it not like `jest` and `vitest.` To solve this you will have to use try catch block but obviously what we really want is that `bun:test` will really be `jest/vitest` compatible.

**Comment by lucasmonstrox** [comment]
Same problem here!

**Comment by moderj** [comment]
Fixed in v1.0.18
